### PR TITLE
[DE-Migration] LPS-125091 buildLang

### PR DESCRIPTION
**Pull request by ealonso**
Hey guys,

I've added a new tab for user personal folder on upload field. Now, It's look like this:

New_Web_Content_-_Liferay

I tried to keep the same behaviour that we have, I just copied some logic from DL. Also, I copied some logic from Journal Field, since we have also two tabs.

Rergarding, to the tab name, I've use, the "User Personal Folder" as tab name, since reading the https://issues.liferay.com/browse/LPS-122339, I though that it is a good name for the tab, please feel free to change it if it not fit for you.

Please let me know if you have any question.

Regards,
Eudaldo.

### CODE REVIEW

**Comment by ealonso** [comment]
ci:test:sf

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by jeyvison** [comment]
@natocesarrego These changes are basically on Forms. Do you wanna take a look?

**Comment by ealonso** [comment]
I need to update the tests

**Comment by ealonso** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/item/selector/DDMUserPersonalFolderItemSelectorView.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+		servletRequest.setAttribute(
+			DDMUserPersonalFolderItemSelectorViewDisplayContext.class.getName(),
+			ddmUserPersonalFolderItemSelectorViewDisplayContext);
+
+		ServletContext servletContext = getServletContext();
+
+		RequestDispatcher requestDispatcher =
+			servletContext.getRequestDispatcher(
+				"/item_selector/ddm_user_personal_folder.jsp");
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/item/selector/DDMUserPersonalFolderItemSelectorView.java:105]: @ealonso, usually, we don't use the prefix `ddm` to name the JSP files, do you mind to remove it and keep the name only as `user_personal_folder.jsp`?

**Review by natocesarrego — COMMENTED** [review]
Good job, @ealonso! 😄 I just did two little comments regarding naming, can you please check them?

Additionally, I was in doubt if we really should create a new module (`dynamic-data-mapping-form-item-selector-api`) just to hold the class `DDMUserPersonalFolderItemSelectorCriterion`. Is this a known pattern on portal?

Maybe, we can create just a new Java package on the existing module `dynamic-data-mapping-api`, what do you think?

Gracias! 🇪🇸 🙂

**Comment by ealonso** [comment]
Hey @natocesarrego,

Regarding the creation of new module, we usually have the ItemSelectorCriterion in a *-item-selector-api module in portal. Take a look at:

* FragmentItemSelectorCriterion
* CommerceChannelItemSelectorCriterion
* UserItemSelectorCriterion
* SiteNavigationMenuItemItemSelectorCriterion

Regarding the renames, It is confusing for me since you have the following classes in different packages on web module:

* DDMFormUploadFileEntryHandler
* DDMFormUploadResponseHandler
* DDMFormAdminDisplayContext
* DDMFormAdminFieldSetDisplayContext
* DDMFormDDMDataProviderDisplay
* DDMFormBrowserPortlet
* DDMFormBrowsePortletProvider
* DDMFormInstaceSearchTerms
...

So for that reason I decided to prefixed everything. If I removed the DDM prefix for one place, I should rename all the classes on my pull to make it consistent. What do you think? Should I rename everything removing the DDM prefix?

Regards,
Eudaldo

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/UploadDDMUserPersonalFolderMVCActionCommand.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Eudaldo Alonso
+ */
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM,
+		"mvc.command.name=/dynamic_data_mapping_form/upload_ddm_user_personal_folder"
```
[the remark this answers is not in the document]
Comment (ealonso) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/UploadDDMUserPersonalFolderMVCActionCommand.java:37]: @natocesarrego Could you please give me an example?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/item/selector/DDMUserPersonalFolderItemSelectorView.java (hunk 1 above)
Comment (ealonso) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/item/selector/DDMUserPersonalFolderItemSelectorView.java:105]: Done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/UploadDDMUserPersonalFolderMVCActionCommand.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/UploadDDMUserPersonalFolderMVCActionCommand.java:37]: Nevermind, @ealonso, I confused with the DDM Form **Admin** Portlet. 😁 This way is fine. 🙂 Thanks.

**Comment by natocesarrego** [comment]
I've checked the code, @ealonso, and I've realized that in some places a new module is created and in other places it's not, because of that I've asked if this is a pattern along portal. 🙂 IMHO, if this is a pattern, then let's follow it, otherwise I understand that create a module just to hold a class is too much.

cc @jeyvison

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by jeyvison** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
Hey @jeyvison @natocesarrego,

Can we forward it? or Do you want any other change? 

Thanks!
Eudaldo

**Comment by jeyvison** [comment]
ci:forward

**Pull request closed without merging**